modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/test/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/DateRangeFunctionTest.java

```diff
@@ -82,6 +80,120 @@ public void testApplyFalse2() {
 				).toString()));
 	}
 
+	@Test
+	public void testApplyFalse3() {
+		Assert.assertFalse(
+			_dateRangeFunction.apply(
+				null,
+				JSONUtil.put(
+					"endsOn",
+					JSONUtil.put(
+						"date", "responseDate"
+					).put(
+						"quantity", -12
+					).put(
+						"type", "customDate"
+					).put(
+						"unit", "days"
+					)
+				).put(
+					"startsFrom",
+					JSONUtil.put(
+						"date", "responseDate"
+					).put(
+						"quantity", -24
+					).put(
+						"type", "customDate"
+					).put(
+						"unit", "days"
+					)
+				)));
+	}
+
+	@Test
+	public void testApplyFalseCustomDays1() {
+		LocalDate localDate = _todayLocalDate.plusDays(6);
+
+		Assert.assertFalse(_apply(localDate, "days", 12, 24));
```

Comment:
@mlan0 How about inline this and create only one test called `testApplyFalseCustomDays`?

```
public void testApplyFalseCustomDays() {
     Assert.assertFalse(_apply(_todayLocalDate.minusDays(6), "days", -24, -12));
     Assert.assertFalse(_apply(_todayLocalDate.minusDays(30), "days", -24, -12));
     Assert.assertFalse(_apply(_todayLocalDate.plusDays(6), "days", 12, 24));
     Assert.assertFalse(_apply(_todayLocalDate.plusDays(30), "days", 12, 24));
}
```
Note that I sorted alphabetically putting `minusDays` before `plusDays`.

This same comment can be applied to `testApplyFalseCustomMonths`, `testApplyFalseCustomYears` and `testApplyTrue...`.

Reply:
That's a great idea. I changed it.